Article first draft (#135)
hassanobeid1994/tr_b_causal_2020: article/sections/_3_graph_overview.tex

\section{Overview of Causal Graphs}
\label{sec:graph-overview}

> Causal diagrams and causal graphical models have been introduced by \citet{pearl_causality_2000} as a powerful tool for causal inference, especially in observational studies.

Author: ```suggestion
Causal diagrams and causal graphical models have been introduced by \citet{pearl_1995_causal} as a powerful tool for causal inference, especially in observational studies.
```

Author: Also, could you link back to section 2? At the moment, the link between sections 2 and 3 are quite jarring because neither section hints or makes reference to the other section's existence.

Perhaps one of the most important and useful features of causal
graphs when dealing with causal inference problems is the clear illustration
of the causal relationships between the variables.

> While a formal introduction to the topic of directed acyclic graphs (DAGs) is beyond the scope of this chapter,

Author: Note that DAG is defined in section 1.
```suggestion
While a formal introduction to DAGs is beyond the scope of this chapter,
```

here we focus specifically on illustrating the power of DAGs to represent

> and encode complex  causal relationships between variables in an intuitive and

Author: ```suggestion
and encode complex causal relationships between variables in an intuitive and
```

clear manner.
Interested readers can refer to \citet{pearl_causality_2000} for a thorough introduction.

Consider the causal graph represented in Figure \ref{fig:simple-graph}.

> Suppose we're interested in the effect of Z on Y.

Author: ```suggestion
Suppose we're interested in the effect of a treatment Z on Y.
```

What the graph in Figure \ref{fig:simple-graph} implies is that Z is

> independent of Y(Z) given X, or in other words, the assignment mechanism is

Author: ```suggestion
independent of Y(Z, X) given X. In causal jargon, the treatment assignment mechanism is
```

Author: Hassan, the way I read (and understand) this sentence is as follows.

"What the graph in Figure \ref{fig:simple-graph} implies is that Z is generated before and in ignorance of Y. Statistically, this implies that Z is independent of Y(Z, X). In causal jargon, the mechanism by which the treatment (Z) was assigned is ignorable, conditional on the values of X."

For people who are not already familiar with causal inference, I think this may be a clearer expression of your idea. I think they may get lost at "assignment mechanism" if they don't really know what is being assigned.

> ignorable conditional on the values of X.

Author: ```suggestion
ignorable, conditional on the values of X.
```

In such situations, it is sufficient to control for
X to obtain an unbiased estimate of the causal effect of Z on Y.

\begin{figure}[h!]
   \centering
   \includegraphics[height=0.5\textwidth]{simple-graph.pdf}
   \caption{Simple DAG of the causal relationships between X, Y and Z}
   \label{fig:simple-graph}
\end{figure}

In comparison, note the set of structural equations needed to convey the same assumptions as Figure \ref{fig:simple-graph}.
\[Z = f_Z(X, \epsilon_Z)  \]

> \[Y(z) = f_Y(X, z, \epsilon_Y)  \]

Author: I think more equations are needed. For instance, in order to know that Z and Y are independent, we need \epsilon_Z and \epsilon_Y to be independent. We also need `X = f_X ( \epsilon_X )`

Now consider the case where there exists another latent confounding variable, U, which also affects both the treatment Z, as well as the outcome, Y.
Figure \ref{fig:simple-graph-confounded} illustrates this assumption in DAG form, and the equations below are the structural equation equivalent:

> \[Z = f_Z(X, U, \epsilon_Z)  \]

Author: I think you need at least three more equations.
X = f_X( \epsilon_X )
U = f_U ( \epsilon_U )
\epsilon_X independent \epsilon_U independent \epsilon_Z independent \epsilon_Y

\[Y(z) = f_Y(X, U, z, \epsilon_Y)  \]

\begin{figure}[h!]
   \centering
   \includegraphics[height=0.5\textwidth]{simple-graph-confounded.pdf}
   \caption{Simple DAG of the causal relationships between X, Y, Z and confounder U}
   \label{fig:simple-graph-confounded}
\end{figure}

> It is clear from Figure \ref{fig:simple-graph-confounded} that conditioning on X alone does not block all the indirect paths between Z and Y, and thus ommitting it will yield biased results of the causal relationship between Z and Y.

Author: In this line, you speak of conditioning on X, but you haven't defined what it means to condition on X. I think you should define the term before using it!

Also, this concept is also subtler than meets the eye. Conditioning on X is NOT equal to adjusting / controlling for X. See for example the definition of conditioning on page 3 [here](https://escholarship.org/content/qt3ng2r0sm/qt3ng2r0sm.pdf).

Basically, conditioning = stratifying to a given level of a covariate, and adjusting / controlling for = including X in one's analysis. One way to see the difference is taking a given choice model, then estimating that model for all observations with say X = 1. You may get all sorts of coefficients for covariates A, B, and C. However, if you estimate a model with all observations and you include a dummy variable for X = 1 it's not the same since you're leaving out the interactions of (X = 1) * A, (X = 1) * B, and so forth that would let the coefficients equal each other across the two models.

Author: ```suggestion
Figure \ref{fig:simple-graph-confounded} shows an indirect connection between Z and Y that goes through U.
Accordingly, even if we condition on X, omitting U will yield biased results of the causal relationship between Z and Y.
```

Author: Also, it's frequently the case that at least for some people, "it is clear" statements will be unclear. I've found it useful to instead replace "it is clear" with a statement that makes clear what I want a reader to see.

Author: Last thought on this, I also think people won't understand your statement about blocking the indirect paths. Grokking your sentence requires one to know that when evaluating a backdoor path, we can go from Y -> U -> Z when U is not conditioned on, but we cannot when U is conditioned on. I wouldn't have known this without taking STAT 281a (or coming across the knowledge via detailed and kind of esoteric study).

> The structural equations also show that the ignorability assumption

Author: How exactly do the equations show this? Or rather, what about the equations shows this?

Author: I assume the reader won't get it automatically. I think you want to point out the existence of U in the equation for Z and Y. Then you essentially write that Z(U) not independent Y(X, U, Z) | X.

of Z does not hold if we only condition on X, and we risk obtaining biased estimates of the causal effect of Z on Y if we fail to
account for U.

> Even in this very simple example with only three or four variables, one can see the advantage of using a causal graph to encode assumptions.

Author: ```suggestion
Even in this very simple example with only three or four variables, we gain an advantage in parsimony and expressiveness by using a causal graph to encode assumptions.
```

> This advantage of DAGs becomes even more apparent in larger problems with many covariates available, and where the causal structure of the data is way more complicated, which are the types of problems typically faced by transportation demand modelers.

Author: ```suggestion
The conciseness of DAGs becomes even more useful in real-world applications.
Here, applied problems may have many available covariates and a complex causal structure.
In such problems, fully writing out the structural equations for the system may be tedious, and readers may still find it hard to understand.
```

> Another benefit of DAGs is that they come with a set of testable implications, and incorporating them in any causal
> analysis adds robustness and defensibility to one's analysis.

Author: ```suggestion
Another benefit of DAGs is that they come with testable implications.
Incorporating these tests in any causal analysis adds robustness and defensibility to one's study.
Conversely, failing to examine these implications signals a lack of thoroughness and invites skepticism of one's results.
```

> We discuss those implications in section 5, and illustrate how to use those tests in one's analysis.

Author: ```suggestion
In section 5, we discuss these implications further, and we illustrate how to tests for them in one's analysis.
```

> Lastly, it is important to note that the graphical approach to causality focuses
> primarily on issues of identification of causal effects, that is, given a
> directed acyclic graph (DAG) that encodes an analyst's knowledge and belief
> about the data generation process of the problem at hand, can a specific
> causal effect be identified?

Author: ```suggestion
Lastly, it is important to note that the graphical approach to causality focuses
primarily on issues of identification of causal effects. That is, given a
directed acyclic graph (DAG) that encodes an analyst's knowledge and belief
about the data generation process of the problem at hand, can a specific
causal effect be identified?
```

As such, we emphasize that DAGs are great tools

> for a modeler to encode their assumptions about a problem, but not necessarily
> a guide on how to estimate a causal effects of interest.

Author: ```suggestion
for a modeler to encode their assumptions about a problem, even though we will use more familiar statistical tools to actually infer/estimate a causal effects of interest.
```

\subsection{Prior Uses of Causal Graphs in Choice Modelling}
\label{sec:choice-graphs}

In our last subsection, we reviewed the basics of causal graphs: what are they and why are they useful?
We targeted that subsection at choice modellers who are unfamiliar with these tools.
However, readers should be aware of the history of causal graphs in choice modelling.
Indeed, choice modellers have used causal graphs (in limited fashion) for years.
Here are three examples to illustrate our point.
First, consider the case of Random Utility Maximization (RUM) models.

\begin{figure}
   \centering
   \includegraphics[width=0.25\textwidth]{rum-causal-graph}
   \caption{Archetypical RUM causal diagram}
   \label{fig:example-graph-rum}
\end{figure}

For decades, choice modellers have used stylized DAGs to depict RUM models.

> In particular, these diagrams illustrate an assumed choice-making (i.e., causal) process.

Reviewer: Maybe drop the (i.e., causal)

Author: Done!

As an example, see Figure \ref{fig:example-graph-rum}, reproduced from Figure 1 of \citet{ben_2002_integration}.
\citeauthor{ben_2002_integration} draw a two-part process.
First, they assume that explanatory variables cause unobserved utilities.
Then, they assume that the unobserved utilities cause the choice.

Note that although RUM diagrams adequately show the choice process, we still call them stylized.
Specifically, these diagrams lack detail about the relationships between explanatory variables.
When speaking collectively, we cannot tell if one explanatory variable causes another.

> Unfortunately, as shown in Section \ref{sec:graph-importance}, such knowledge is crucial.

Reviewer: Let's make sure in the graph-importance section that we emphasize the common practice of conditioning on post-treatment variables, which makes causal graphs all the more important to help think about the structural equations.

Without more detailed causal knowledge, our inferences may be inconsistent and arbitrarily bad.

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{iclv-causal-graph}
   \caption{Archetypical ICLV causal diagram}
   \label{fig:example-graph-iclv}
\end{figure}

Besides RUM models, causal graphs appear in the literature on Integrated Choice and Latent Variable (ICLV) models.
An example of an ICLV model is in Figure \ref{fig:example-graph-iclv}, based on Figure 5 of \citet{ben_2002_integration}.
As with RUM models, ICLV causal graphs represent assumptions about the choice process.
Here concern typically centers around unobserved mediators.
These unobserved variables cause the outcome and are caused by one's observed covariates.

> Omitting such mediators leads to models that misrepresent the assumed choice process.

Reviewer: Mediators here seem to be analogous to post-treatment variables in the econometrics literature, which shouldn't be conditioned on, or if conditioned on, further steps are needed. Let's discuss to make sure I'm understanding correctly, and if I am, we can add a small nuance here. 

Please clarify the three different situations that are being discussed.

Author: I think we spoke about the mediators and front-door criterion on Sunday. Let me know if there's anything else of concern in your first paragraph here.

What are the three situations you're referring to? RUM / ICLV / ABM?

Two main features distinguish RUM, ICLV, and ABM graphs from the causal graphs of \citet{pearl_1995_causal}.
First, causal graphs in choice modelling traditionally ignore relations between the explanatory variables.
Typically, choice modelling causal graphs show all explanatory variables together as a monolith.
The relationships between explanatory variables is often not specified.
Even worse, researchers may tacitly treat the variables as if they are jointly independent.
In the language used by a large group of causal inference scholars, econometric causal diagrams ignore the treatment assignment mechanism.

Secondly, causal graphs in choice modelling papers are purely didactic.
They convey how choice modellers perceive the world and the choice generation process.
However, they are seldom treated as a model themselves.
In particular, choice modellers rarely test the predictions of causal graphs.
In doing so, we underuse causal implications such as statistical independence.
Worse, we possibly violate these implications with great frequency.
Such actions ignore the efforts from causal inference in computer science.

> There, researchers stress using their data to test the implications of their causal graphs.

Reviewer: Maybe needs some word-smithing.

Author: Done!
